mypy/modulefinder.py

```diff
@@ -506,21 +506,24 @@ def _find_module(self, id: str, use_typeshed: bool) -> ModuleSearchResult:
             dir_prefix = base_dir
             for _ in range(len(components) - 1):
                 dir_prefix = os.path.dirname(dir_prefix)
+
+            # Stubs-only packages always take precedence over py.typed packages
+            path_stubs = f"{base_path}-stubs{sepinit}.pyi"
```

Comment:
hm this assumes there is an `__init__.pyi`, I don't think that's required?

Reply:
It's pretty much the same as the `path_stubs = base_path + "-stubs" + sepinit + extension` from before before 🤷🏻 

Should I change it to something else?

Reply:
Yea you're right, apparently this isn't the whole story: 
I added some new tests for `a.py[i]`, but those are failing.

Reply:
Hm somehow the tests only fail if it's called `a.pyi`, and not when called `spam.pyi`. I'm guessing it's a caching thing. Either way, it's probably out out scope, and might be limited to these unit-tests.

Reply:
> hm this assumes there is an `__init__.pyi`, I don't think that's required?

See the updated PR description

Reply:
Yeah I believe namespace packages are handled on L525 afterwards.